Google.PowerShell/Provider/GoogleCloudStorageProvider.cs

```diff
@@ -859,13 +861,32 @@ protected override object RemoveItemDynamicParameters(string path, bool recurse)
 
         private void RemoveFolder(GcsPath gcsPath, bool recurse)
         {
+            IEnumerable<Object> children = ListChildren(gcsPath, true);
+            string query = $"The item at the path '{gcsPath.ObjectPath}' has children and the Recurse parameter was not specified."
+                + " If you continue, all children will be removed with the item. Are you sure you want to continue?";
+            string caption = "Confirm";
+
+            // Ask the user for permission to delete children of the folder if -Recurse is not supplied.
+            if (children.Any() && !recurse)
+            {
+                if (ShouldContinue(query, caption))
+                {
+                    recurse = true;
+                }
+                else
+                {
+                    return;
```

Comment:
Even if we don't delete all of the children, we should try to delete the folder object if it is real. It could contain undesired data and there is already a ShouldProcess gate on it.